### PR TITLE
Add support for SSL_CIPHER

### PR DESCRIPTION
**Pull request by erikjohnston**
Also, adds SSL_get_version

### CODE REVIEW

File: openssl/src/ssl/mod.rs [hunk 1]
```diff
@@ -769,6 +769,67 @@ impl SslContext {
...
+        str::from_utf8(version.to_bytes()).unwrap()
+    }
+
+    /// Returns the number of secret bits used for the cipher.
+    ///
+    /// The first element is the number of secret bits used for the cipher.
+    ///
+    /// The second element, if not None, is the number of bits processed by
+    /// the chosen algorithm,
+    pub fn bits(&self) -> (i32, Option<i32>) {
```
Comment (sfackler) [inline openssl/src/ssl/mod.rs]: Could you make this return a struct instead? Something as simple as

```rust
pub struct CipherBits {
    pub secret: i32,
    pub algorithm: Option<i32>,
}
```

Comment (erikjohnston) [inline openssl/src/ssl/mod.rs]: Yes, that would be much nicer.

**Comment by erikjohnston** [comment]
I should also add that I've assumed the `SSL_CIPHER` returned by `SSL_get_current_cipher` is only valid for the lifetime of the `SSL` object, but I couldn't actually find anything in the openssl docs about it.

**Comment by sfackler** [comment]
Yep, OpenSSL's documentation is pretty awful :) I'll jump into the source tonight and make sure, but I'd guess your assumption is correct.

**Comment by erikjohnston** [comment]
Yes it is awful :)

Thanks, I managed to double check that all the static strings are in fact static strings by looking at the source, but then I started running a fever and collapsed (for, like, a week). I've only just recovered, so I'm quite happy to leave it to you to check that last detail!

**Pull request merged**